src/main/java/com/eerussianguy/blazemap/engine/client/BlazeMapClientEngine.java

```diff
@@ -159,6 +162,19 @@ public static void onChunkChanged(ChunkPos pos, String source) {
     public static void submitChanges(ResourceKey<Level> dimension, ChunkPos pos, List<MasterDatum> data, String source) {
         isServerSource = true;
         mdSource = source;
+
+        // If the storage hasn't been set, that means the player login event hasn't been received yet
+        // and getPipeline will throw a NullPointerException that crashes the game.
+        // To avoid this, will wait until storage exists before submitting this MD
+        while (storage == null) {
+            try {
+                Thread.sleep(50); // Approx 1 tick wait
+            } catch (InterruptedException e) {
+                // This isn't supposed to happen, but I can't compile without it
+                e.printStackTrace();
+            }
+        }
```

Comment:
As far as I'm aware, this only affects the state when the internal server is active, which it was at the time I made the commit for this. Will probably change more/make nicer as part of the follow up with internal server enabled along with the rest of that refactor to make it behave nicer async.

Reply:
This is problematic because you're causing a delay in the game's network thread, which stalls everything.

Reply:
Ahhh, there goes me thinking in async TS terms again where, when the thread sleeps in one place, it goes off to do something else in the mean time until the timer is up!

The simple alternative here is to just do a null check as an if statement, however, I found that then those chunks wouldn't load on the map at all until another update happened to occur. Which, for a single player world or anyone loading into not-yet-generated chunks, probably won't take too long, but might not happen naturally in a pre-genned or multiplayer (so naturally pregenned by another player) world that's already settled into a "steady state".

I think I'm gonna have to think of a way to add some secondary queue to store the data until storage != null and then make all the updates in one go once that's the case, at least until we decide it's time to reenable the internal server and I can do the refactor properly. That's probably the simplest solution here for the time being.

Speaking of said refactor, will leave it to you wether or not you want to prioritise getting the server available again or getting 1.20.1 working (should probably remove the sever option entirely in 1.19+ if this is the case). Am happy to move focus to either, though have continued on the version upgrading path in the mean time given the server's already partially blocked by BME-13 and it'll probably be easier to just do the refactor in the "final" version rather than having to refactor then port said refactor.